tests/xharness/Xharness.Tests/Jenkins/TestSelectorTests.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.DotNet.XHarness.iOS.Shared;
+using Microsoft.DotNet.XHarness.iOS.Shared.Execution;
+using Microsoft.DotNet.XHarness.iOS.Shared.Listeners;
+using Microsoft.DotNet.XHarness.iOS.Shared.Logging;
+using Moq;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+using Xharness.Jenkins;
+
+namespace Xharness.Tests.Jenkins {
+	
+	[TestFixture]
+	public class TestSelectorTests {
+		Xharness.Jenkins.Jenkins jenkins;
+		Mock<IResultParser> resultParser;
+		Mock<ITunnelBore> tunnelBore;	
+		Mock<IHarness> harness;
+		Mock<IProcessManager> processManager;
+		Mock<IVersionControlSystem> versionControlSystem;
+		Mock<ILog> log;
+		TestSelector testSelector;
+		string tmpPath;
+		
+		[SetUp]
+		public void SetUp ()
+		{
+			tmpPath = Path.GetTempPath ();
+			harness = new Mock<IHarness> (); 
+			resultParser = new Mock<IResultParser> ();
+			tunnelBore = new Mock<ITunnelBore> ();
+			processManager = new Mock<IProcessManager> ();
+			versionControlSystem = new Mock<IVersionControlSystem> ();
+			log = new Mock<ILog> ();
+
+			// defult expectations
+			harness.Setup (h => h.ENABLE_XAMARIN).Returns (true);
+			harness.Setup (h => h.JENKINS_RESULTS_DIRECTORY).Returns (tmpPath);
+			harness.Setup (h => h.Labels).Returns (new HashSet<string> ());
+			harness.Setup (h => h.XcodeRoot).Returns ("Xcode11.app");
+
+			jenkins = new Xharness.Jenkins.Jenkins (harness.Object, processManager.Object, resultParser.Object,
+				tunnelBore.Object);
+			jenkins.MainLog = log.Object;
+			testSelector = new TestSelector (jenkins, processManager.Object, versionControlSystem.Object);
+		}
+
+		[TearDown]
+		public void TearDown ()
+		{
+			processManager = null;
+			versionControlSystem = null;
+			harness = null;
+			jenkins = null;
+			testSelector = null;
+			Environment.SetEnvironmentVariable ("BRANCH_NAME", null);
+		}
+
+
+		public class TestCasesData {
+			public static IEnumerable SelectByLabelTestData {
+				get {
+					// create the test data to ensure that the correct tests are selected according to the labels
+
+					// all tests selected, not main branch
+					TestSelection expectedSelection = TestSelection.All;
+					yield return new TestCaseData ("testBranch", new List<string> {"run-all-tests"}, expectedSelection, true);
+
+					// all tests selected via tag, but we added some skips, therefore we skip them
+					expectedSelection = TestSelection.All;
+					expectedSelection &= ~TestSelection.MacOS;
+					yield return new TestCaseData ("testBranch", 
+						new List<string> { "run-all-tests", "skip-mac-tests"},
+						expectedSelection, true);
+					
+					// skip all ye select one of the tests
+					expectedSelection = TestSelection.MacOS;
+					yield return new TestCaseData ("testBranch",
+						new List<string> {"skip-all-tests", "run-mac-tests"},
```

Comment:
There are several places that are missing spaces:

```suggestion
						new List<string> { "skip-all-tests", "run-mac-tests" },
```